scalameter-core/src/main/scala/org/scalameter/InvocationCounting.scala

```diff
@@ -0,0 +1,64 @@
+package org.scalameter
+
+import java.io.File
+import scala.collection.Seq
+import org.scalameter.execution.invocation.InvocationCountMatcher
+import org.scalameter.execution.invocation.instrumentation.{Instrumentation, MethodInvocationCounter}
+import org.scalameter.Key._
+
+
+/** Mixin for all [[org.scalameter.Measurer]] implementations that perform any kind of method invocation counting. */
+trait InvocationCounting extends Measurer {
+  def matcher: InvocationCountMatcher
+
+  def units: String = "#"
+
+  def measure[T, U](context: Context, measurements: Int, setup: (T) => Any, tear: (T) => Any, regen: () => T, snippet: (T) => Any): Seq[Double] = {
+    var iteration = 0
+    val invocationsBuilder = List.newBuilder[Double]
+    var value: T = null.asInstanceOf[T]
+    var obj: Any = null.asInstanceOf[Any]
+    val numMethods = context.goe(exec.measurers.methodInvocationLookupTable,
+      sys.error("Measurer.prepareContext must be called before Measurer.measure")).length
+
+    while (iteration < measurements) {
+      value = regen()
+
+      MethodInvocationCounter.setup(numMethods)
+      setup(value)
+
+      MethodInvocationCounter.start()
+      obj = snippet(value)
+      MethodInvocationCounter.stop()
+      tear(value)
+
+      invocationsBuilder += MethodInvocationCounter.counts.sum
+      iteration += 1
+    }
+
+    val invocations = invocationsBuilder.result()
+    log.verbose("Measurements: " + invocations.mkString(", "))
+    invocations
+  }
+
+  /** Creates a jar with instrumented classes, prepends it to a classpath key and saves the internal method lookup table. */
+  override def prepareContext(context: Context): Context = {
+    val jar = File.createTempFile(s"scalameter-bench-", "-instrumented.jar")
+    jar.deleteOnExit()
```

Comment:
Since `prepareContext` is called at the beginning of the run, before the separate VM is started, this file is not deleted until you exit the host VM. Meaning that if you call the test several times from SBT, these will leak.
You should probably delete the file manually (in addition to this call) at the end of the `run` method in the `Executor`.